### PR TITLE
Refactor native code on Android

### PR DESCRIPTION
**Pull request by pklatka**
## Summary
This PR is a follow-up to #57: refactors the Android native code and simplifies the message object format (no longer need to wrap the object with `{message: ...}`)

### CODE REVIEW

File: packages/turbo/ios/RNVisitableView.swift [hunk 1]
```diff
@@ -95,6 +91,7 @@ extension RNVisitableView: RNVisitableViewControllerDelegate {
   
   func visitableWillAppear(visitable: Visitable) {
     session.registerVisitableView(newView: self)
+    visit()
```
Comment (pklatka) [inline packages/turbo/ios/RNVisitableView.swift:94]: Moving visit() here to match Android code.

File: packages/turbo/android/src/main/java/com/reactnativeturbowebview/RNSession.kt [hunk 2]
```diff
@@ -1,102 +1,156 @@
...
+    visitableViews.last().visitLocationStarted(location)
+  }
+
+  override fun visitProposedToLocation(location: String, options: TurboVisitOptions) {
+    visitableViews.last().visitProposedToLocation(location, options)
+  }
+
+  override fun visitRendered() {
+    visitableViews.last().visitRendered()
+  }
```
Comment (pfeiffer) [inline packages/turbo/android/src/main/java/com/reactnativeturbowebview/RNSession.kt:152]: Could we end up with `visitableViews.last()` returning `null` in the case that a VisitableView disappears and thus is unregistered before a visit is finished? In that case, would we call `visitCompleted()` on a null object?

File: packages/turbo/android/src/main/java/com/reactnativeturbowebview/RNSession.kt [hunk 3]
```diff
@@ -1,102 +1,156 @@
...
-
-      for (view in registeredVisitableViews) {
-        view.detachWebView() {
-          onDetached()
-        }
-      }
-    }
-
-    if (!registeredVisitableViews.contains(newView)) {
-      registeredVisitableViews.add(newView)
```
Comment (pfeiffer) [inline packages/turbo/android/src/main/java/com/reactnativeturbowebview/RNSession.kt:67]: Wow, great to see how readable this becomes after the refactor 🎉

File: packages/turbo/android/src/main/java/com/reactnativeturbowebview/RNSession.kt [hunk 4]
```diff
@@ -1,102 +1,156 @@
 package com.reactnativeturbowebview
```
Comment (pfeiffer) [inline packages/turbo/android/src/main/java/com/reactnativeturbowebview/RNSession.kt:1]: Should we consider renaming this to be more in-line with the package name(s)?

File: packages/turbo/src/VisitableView.tsx [hunk 5]
```diff
@@ -105,14 +105,11 @@ const VisitableView = React.forwardRef<RefObject, React.PropsWithRef<Props>>(
...
-      (e: NativeSyntheticEvent<{ message: object }>) => {
-        onMessageCallbacks.current.forEach((listener) => {
-          listener?.(e.nativeEvent);
-        });
-      },
-      []
-    );
+    const handleOnMessage = useCallback((e: NativeSyntheticEvent<object>) => {
+      onMessageCallbacks.current.forEach((listener) => {
+        listener?.(e.nativeEvent);
```
Comment (mironiasty) [inline packages/turbo/src/VisitableView.tsx:110]: note for myself: we can remove nullability for listeners to simplify code 🤔

File: packages/turbo/android/src/main/java/com/reactnativeturbowebview/RNSession.kt (hunk 4 above)
Comment (pklatka) [reply packages/turbo/android/src/main/java/com/reactnativeturbowebview/RNSession.kt:1]: As discussed earlier, we will probably rename the whole library so let's leave it for now.

**Review by mironiasty — APPROVED** [review]

**Pull request merged**